test/registry.test.ts

```diff
@@ -208,22 +227,217 @@ describe("BeranamesRegistry", function () {
+
                 await expect(registry.mintERC20(
                     ["o", "o", "g", "a"],
                     1,
                     owner.address,
                     "https://www.google.com",
                     owner.address,
-                    manager.address,
+                    erc20.address,
                 )).to.not.be.reverted;
             });
         });
+
+        describe("renewERC20", function () {
+            it("Should allow to renew ERC20", async function () {
+                const { registry, owner, erc20 } = await loadFixture(setupFixture);
+                await registry.togglePause();
+                await erc20.approve(registry.address, BigInt("90000000000000000000"));
+                await erc20.mint(BigInt("90000000000000000000"), owner.address);
+
+                await registry.mintERC20(
+                    ["o", "o", "g", "a"],
+                    1,
+                    owner.address,
+                    "https://www.google.com",
+                    owner.address,
+                    erc20.address,
+                );
+
+                await network.provider.send("evm_increaseTime", [86400 * (365 + 30)]);
+
+                await erc20.approve(registry.address, BigInt("90000000000000000000"));
+
+                await expect(registry.renewERC20(
+                    ["o", "o", "g", "a"],
+                    1,
+                    erc20.address,
+                )).to.not.be.reverted;
+            });
+            it("Should not allow to renew if name is not minted", async function () {
+                const { registry, erc20 } = await loadFixture(setupFixture);
+                await registry.togglePause();
+                await erc20.approve(registry.address, BigInt("90000000000000000000"));
+
+                await expect(registry.renewERC20(["o", "o", "g", "a"], 1, erc20.address))
+                    .to.be.revertedWithCustomError(registry, "NoEntity");
+            });
+        });
+
+        describe("updateWhois", function () {
+            it("Should not allow to update whois if sender is not an owner of token", async function () {
+                const { registry, owner, otherAccount } = await loadFixture(setupFixture);
+
+                const chars = ["o", "o", "g", "a"];
+
+                await registry.togglePause();
+                await registry.mintNative(
+                    chars,
+                    1,
+                    owner.address,
+                    "www.google.com",
+                    owner.address,
+                    {
+                        value: parseEther("80"),
+                    }
+                );
+
+                // just logged this num from the registry
+                const id = BigInt("68555854115171307888772731598165398131425019690431209975676620582509999362523");
+                // This gives another id, reckon it's a difference in solidity encoding and ethers
```

Comment:
idk maybe you know how to encode and get the same value as solidity does(

Reply:
could try to use https://docs.ethers.org/v5/api/utils/hashing/#utils-keccak256 and wrapping it in a BigInt()